src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -828,6 +789,64 @@ class npc_salanar_the_horseman : public CreatureScript
     }
 };
 
+class spell_deliver_stolen_horse : public SpellScriptLoader
+{
+public:
+    spell_deliver_stolen_horse() : SpellScriptLoader("spell_deliver_stolen_horse") { }
+
+    class spell_deliver_stolen_horse_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_deliver_stolen_horse_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_DELIVER_STOLEN_HORSE, SPELL_EFFECT_STOLEN_HORSE });
+        }
+
+        void HitSalanarCheck(SpellEffIndex /*effIndex*/)
+        {
+            // Can only hit Salanar due to TARGET_UNIT_NEARBY_ENTRY
+            if (Unit* salanar = GetHitUnit())
+                _salanarNear = true;
+        }
+
+        void HitTargetCheck(SpellEffIndex effIndex)
+        {
+            if (!_salanarNear)
+            {
+                PreventHitEffect(effIndex);
+                return;
+            }
+
+            if (Player* player = GetHitPlayer())
+                player->RemoveAurasDueToSpell(SPELL_EFFECT_STOLEN_HORSE);
+
+            if (Unit* caster = GetCaster())
+            {
+                caster->RemoveAurasDueToSpell(SPELL_EFFECT_STOLEN_HORSE);
+
+                caster->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_SPELLCLICK);
+                caster->SetFaction(FACTION_FRIENDLY);
+
+                caster->CastSpell(caster, SPELL_CALL_DARK_RIDER, true);
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_deliver_stolen_horse_SpellScript::HitSalanarCheck, EFFECT_0, SPELL_EFFECT_DUMMY);
+            OnEffectHitTarget += SpellEffectFn(spell_deliver_stolen_horse_SpellScript::HitTargetCheck, EFFECT_1, SPELL_EFFECT_KILL_CREDIT2);
+        }
+
+        bool _salanarNear;
```

Comment:
it's not initialized
Use this hook to initialize it: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellScript.h#L136-L138